### ISSUE TITLE
Feature Request: Liveness Probe on API deployment

### ISSUE THREAD

**Issue by dmckernanacv** [reportportal/kubernetes#268]
Hello, I would like to request the addition of liveness probes (and actually all probes would be good) on the [API deployment](https://github.com/reportportal/kubernetes/blob/develop/reportportal/templates/api-deployment.yaml) resource (All deployments could benefit from these as well).

Perhaps these could be optional, controlled via a value?

`/reportportal/values.yaml`
```diff 
serviceapi:
  readinessProbe:
+   enabled: true
    initialDelaySeconds: 30
    periodSeconds: 20
    timeoutSeconds: 3
    failureThreshold: 20
+ livenessProbe:
+   enabled: true
+   initialDelaySeconds: 30
+   periodSeconds: 20
+   timeoutSeconds: 3
+   failureThreshold: 5
```

`/reportportal/templates/api-deployment.yaml`
```diff 
spec:
  template:
    spec:
      containers:
        # ........
+       {{- if .Values.serviceapi.readinessProbe.enabled }}
        readinessProbe:
          httpGet:
            path: "{{ .Values.rp.healthEndpoint }}"
            port: 8585
          initialDelaySeconds: {{ .Values.serviceapi.readinessProbe.initialDelaySeconds }}
          periodSeconds: {{ .Values.serviceapi.readinessProbe.periodSeconds }}
          timeoutSeconds: {{ .Values.serviceapi.readinessProbe.timeoutSeconds }}
          failureThreshold: {{ .Values.serviceapi.readinessProbe.failureThreshold }}
+       {{- end }}
+       {{- if .Values.serviceapi.livenessProbe.enabled }}
+       livenessProbe:
+         httpGet:
+           path: "{{ .Values.rp.healthEndpoint }}"
+           port: 8585
+         initialDelaySeconds: {{ .Values.serviceapi.livenessProbe.initialDelaySeconds }}
+         periodSeconds: {{ .Values.serviceapi.livenessProbe.periodSeconds }}
+         timeoutSeconds: {{ .Values.serviceapi.livenessProbe.timeoutSeconds }}
+         failureThreshold: {{ .Values.serviceapi.livenessProbe.failureThreshold }}
+       {{- end }}
```

This would allow the API (and/or other deployment resources) to automatically restart if they are no longer live (for example, if minio or rabbitmq become unreachable)

**Comment by HardNorth** [comment]
@dmckernanacv Hmm, OK, legit. This is actually due to the nature of Java. It does not lookup for new addresses if it already looked up it. So the only option is to restart JVM. Therefore if a cluster update its nodes, then they become unavailable for service-api until restart.

**Comment by HardNorth** [comment]
Done, will be released in 5.7.3
